core/src/stored/bls.cc

```diff
@@ -130,99 +94,131 @@ int main(int argc, char* argv[])
+
+  bls_app.add_flag("--dt,--debug-timestamp", dbg_timestamp,
+                   "Print timestamp in debug output.");
 
-      case 'k':
-        list_blocks = true;
-        break;
+  FILE* fd;
+  char line[1000];
+  bls_app
+      .add_option(
+          "-e,--exclude",
+          [&line, &fd](std::vector<std::string> val) {
+            if ((fd = fopen(val.front().c_str(), "rb")) == NULL) {
+              BErrNo be;
+              Pmsg2(0, _("Could not open exclude file: %s, ERR=%s\n"), optarg,
+                    be.bstrerror());
+              exit(1);
+            }
+            while (fgets(line, sizeof(line), fd) != NULL) {
+              StripTrailingJunk(line);
+              Dmsg1(100, "add_exclude %s\n", line);
+              AddFnameToExcludeList(ff, line);
+            }
+            fclose(fd);
+            return true;
+          },
+          "Exclude list.")
+      ->type_name("<file>");
+
+  bls_app
+      .add_option(
+          "-i,--include-list",
+          [&line, &fd](std::vector<std::string> val) {
+            if ((fd = fopen(optarg, "rb")) == NULL) {
+              BErrNo be;
+              Pmsg2(0, _("Could not open include file: %s, ERR=%s\n"), optarg,
+                    be.bstrerror());
+              exit(1);
+            }
+            while (fgets(line, sizeof(line), fd) != NULL) {
+              StripTrailingJunk(line);
+              Dmsg1(100, "add_include %s\n", line);
+              AddFnameToIncludeList(ff, 0, line);
+            }
+            fclose(fd);
+            return true;
+          },
+          "Include list.")
+      ->type_name("<file>");
+
+  bool list_jobs = false;
+  bls_app.add_flag("-j,--list-jobs", list_jobs, "List jobs.");
+
+  bool list_blocks = false;
+  bls_app.add_flag("-k,--list-blocks", list_blocks,
+                   "List blocks.\n"
+                   "If neither -j or -k specified, list saved files.");
+
+  bls_app.add_flag("-L,--dump-labels", dump_label, "Dump labels.");
 
-      case 'L':
-        dump_label = true;
-        break;
 
-      case 'p':
+  bool ignore_label_errors = false;
+  bls_app.add_flag(
+      "-p,--ignore-errors",
+      [&ignore_label_errors](bool val) {
         ignore_label_errors = true;
         forge_on = true;
-        break;
+      },
+      "Proceed inspite of IO errors.");
 
-      case 'v':
-        verbose++;
-        break;
+  bls_app
+      .add_flag(
+          "-v,--verbose", [](bool val) { verbose++; }, "Verbose user messages.")
+      ->trigger_on_parse();
 
-      case 'V': /* Volume name */
-        VolumeName = optarg;
-        break;
+  std::string VolumeName;
+  bls_app
+      .add_option("-V,--volume-names", VolumeName,
+                  "Volume names (separated by |)")
+      ->type_name("<vol1|vol2|...>");
 
-      case '?':
-      default:
-        usage();
+  std::vector<std::string> archive_names;
+  bls_app
+      .add_option("archive-names", archive_names, "Specify input archive names")
```

Comment:
You should use the opportunity to better specify this parameter (identical in all the Bareos Volume Utilities), as this has been a source of trouble for a lot of people. Maybe:
```
                  "Specify the input device name (either as name of a Bareos Storage Daemon Device resource or identical to the Archive Device in a Bareos Storage Daemon Device resource .")
```